jmbitcoin/jmbitcoin/secp256k1_transaction.py

```diff
@@ -192,54 +212,63 @@ def segwit_signature_form(txobj, i, script, amount, hashcode=SIGHASH_ALL,
     #                                           amount, hashcode))
     script = decoder_func(script)
-    nVersion = encode(txobj["version"], 256, 4)[::-1]
+    nVersion = struct.pack(b'<I', txobj["version"])
+    if not isinstance(hashcode, int):
+        hashcode = struct.unpack(b'B', hashcode)[0]
     #create hashPrevouts
     if hashcode & SIGHASH_ANYONECANPAY:
-        hashPrevouts = "\x00"*32
+        hashPrevouts = b"\x00"*32
     else:
-        pi = ""
+        pi = b""
         for inp in txobj["ins"]:
             pi += decoder_func(inp["outpoint"]["hash"])[::-1]
-            pi += encode(inp["outpoint"]["index"], 256, 4)[::-1]
+            pi += struct.pack(b'<I', inp["outpoint"]["index"])
         hashPrevouts = bin_dbl_sha256(pi)
     #create hashSequence
     if not hashcode & SIGHASH_ANYONECANPAY and not (
         hashcode & 0x1f == SIGHASH_SINGLE) and not (hashcode & 0x1f == SIGHASH_NONE):
-        pi = ""
+        pi = b""
         for inp in txobj["ins"]:
-            pi += encode(inp["sequence"], 256, 4)[::-1]
+            pi += struct.pack(b'<I', inp["sequence"])
         hashSequence = bin_dbl_sha256(pi)
     else:
-        hashSequence = "\x00"*32
+        hashSequence = b"\x00"*32
     #add this input's outpoint
     thisOut = decoder_func(txobj["ins"][i]["outpoint"]["hash"])[::-1]
-    thisOut += encode(txobj["ins"][i]["outpoint"]["index"], 256, 4)[::-1]
+    thisOut += struct.pack(b'<I', txobj["ins"][i]["outpoint"]["index"])
     scriptCode = num_to_var_int(len(script)) + script
-    amt = encode(amount, 256, 8)[::-1]
-    thisSeq = encode(txobj["ins"][i]["sequence"], 256, 4)[::-1]
+    amt = struct.pack(b'<Q', amount)
+    thisSeq = struct.pack(b'<I', txobj["ins"][i]["sequence"])
     #create hashOutputs
     if not (hashcode & 0x1f == SIGHASH_SINGLE) and not (hashcode & 0x1f == SIGHASH_NONE):
-        pi = ""
+        pi = b""
         for out in txobj["outs"]:
-            pi += encode(out["value"], 256, 8)[::-1]
+            pi += struct.pack(b'<Q', out["value"])
             pi += (num_to_var_int(len(decoder_func(out["script"]))) + \
                    decoder_func(out["script"]))
         hashOutputs = bin_dbl_sha256(pi)
     elif hashcode & 0x1f == SIGHASH_SINGLE and i < len(txobj['outs']):
-        pi = encode(txobj["outs"][i]["value"], 256, 8)[::-1]
+        pi = struct.pack(b'<Q', txobj["outs"][i]["value"])
         pi += (num_to_var_int(len(decoder_func(txobj["outs"][i]["script"]))) +
                decoder_func(txobj["outs"][i]["script"]))
         hashOutputs = bin_dbl_sha256(pi)
     else:
-        hashOutputs = "\x00"*32
-    nLockTime = encode(txobj["locktime"], 256, 4)[::-1]
+        hashOutputs = b"\x00"*32
+    nLockTime = struct.pack(b'<I', txobj["locktime"])
     return nVersion + hashPrevouts + hashSequence + thisOut + scriptCode + amt + \
            thisSeq + hashOutputs + nLockTime
 
 def signature_form(tx, i, script, hashcode=SIGHASH_ALL):
```

Comment:
this method will now always return hex-encoded transaction data from what I can tell

Reply:
is that the correct behavior?

Reply:
it was doing the autodetection magic previously